crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -1066,6 +1075,88 @@ impl PackageRange<'_> {
     }
 }
 
+/// Create a range with improved segments for reporting the available versions for a package.
+fn update_availability_range(
+    range: &Range<Version>,
+    available_versions: &BTreeSet<Version>,
+) -> Range<Version> {
+    let mut new_range = Range::empty();
+
+    let (available_range, first_available, last_available) =
+        match (available_versions.first(), available_versions.last()) {
+            // At least one version is available
+            (Some(first), Some(last)) => {
+                let range = Range::<Version>::from_range_bounds((
+                    Bound::Included(first.clone()),
+                    Bound::Included(last.clone()),
```

Comment:
It basically doesn't matter in this case, we're just using it to check if the other range is completely out of bounds.

Reply:
Maybe I should write a note for that, maybe there are some weird cases where the holes do matter?

Reply:
I wrote it this way to avoid doing something like `for version in available_versions { range.contains(version) }` to see if the range contained all or none of the versions.

Reply:
Yeah I think this is cool.

Reply:
Added a note, we probably wont simplify some things that we could.